src/calculatePlayerMoves.ts

```diff
@@ -0,0 +1,243 @@
+    y: startFromHorizontal ? coordinatesPlayer.y : coordinatesSquare.y,
+    x: startFromHorizontal ? coordinatesSquare.x : coordinatesPlayer.x
+  };
+  const playerInBetween = getSquareByCoordinates(coordinatesPlayerInBetween, board);
+  if (playerInBetween.type !== SquareType.Player) throw new Error("Square is not of type 'Player'");
+  if (!playerInBetween.playerId) return false;
+
+  // When wall between the targeted square and previous square is not walkable
+  coordinatesWallInBetween = {
+    y: startFromHorizontal
+      ? coordinatesPlayer.y + (coordinatesSquare.y - coordinatesPlayer.y) / 2 // division by 2, because coordinate difference between player and targeted square is 2 and we select the middle square
+      : coordinatesSquare.y,
+    x: startFromHorizontal ? coordinatesSquare.x : coordinatesPlayer.x + (coordinatesSquare.x - coordinatesPlayer.x) / 2
+  };
+  wallInBetween = getSquareByCoordinates(coordinatesWallInBetween, board);
+  if (wallInBetween.type !== SquareType.Wall) throw new Error("Square is not of type 'Wall'");
+  if (!wallInBetween.isWalkable) return false;
+
+  // When targeted square is not near the border of the board or there is no wall that would block linear jump
+  const LAST_ROW_INDEX = BOARD_WIDTH - 1;
+  if (
+    (startFromHorizontal ? coordinatesSquare.x : coordinatesSquare.y) !== 0 &&
+    (startFromHorizontal ? coordinatesSquare.x : coordinatesSquare.y) !== LAST_ROW_INDEX
+  ) {
+    coordinatesWallInBetween = {
+      y: startFromHorizontal
+        ? coordinatesPlayer.y
+        : coordinatesPlayer.y + ((coordinatesSquare.y - coordinatesPlayer.y) / 2) * 3, // division by 2 and multiplication by 3, because coordinate difference between player and targeted square is 2 and we select the square that is over the targeted square
+      x: startFromHorizontal
+        ? coordinatesPlayer.x + ((coordinatesSquare.x - coordinatesPlayer.x) / 2) * 3
+        : coordinatesPlayer.x
+    };
+    wallInBetween = getSquareByCoordinates(coordinatesWallInBetween, board);
+    if (wallInBetween.type !== SquareType.Wall) throw new Error("Square is not of type 'Wall'");
+    if (wallInBetween.isWalkable) return false;
+  }
+
+  return true;
+};
+
+// To check if player can move diagonally in each direction, two ways must be checked: start checking horizontally and move on to vertically and the opposite.
+const checkDiagonal = (coordinatesSquare: Coordinates, coordinatesPlayer: Coordinates, board: Board): boolean => {
+  if (
+    // Check diagonal starting horizontally, then starting vertically
+    checkDiagonalFromOneSide(coordinatesSquare, coordinatesPlayer, board, true) ||
+    checkDiagonalFromOneSide(coordinatesSquare, coordinatesPlayer, board, false)
+  ) {
+    return true;
+  }
+  return false;
+};
+
+export const updatePlayerMoves = (coordinatesPlayer: Coordinates, board: Board) => {
+  resetMoves(board);
+  // Go through sqaures around the player (both linear and diagonal)

```

Comment:
I am really sorry if this took a lot of work to update the board with `for` cycles, but I do think that hardcoding 12 lines would be much simpler. Make a function like `updatePlayerSquareAvailability(coordinates, validationCallback)` that would check if square exists then call `validationCallback` which are functions like `checkDiagonal` and update the `isAvailable` accordingly. You would call the `updatePlayerSquareAvailability` 12 times